File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        Stack<Character>stack = new Stack<>();
        for(char ch : s.toCharArray()){
            if(ch == ']' && stack.size()>0 && stack.peek() == '['){
                stack.pop();
            }else{
                stack.push(ch);
            }
        }
        return (stack.size()/2+1)/2;
    }
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.cs
public class Solution {
    public int MinSwaps(string s) {
        int brackets = 0;
        for(int i=0;i<s.Length;i++){
            if(s[i] == ']' && brackets > 0){
                brackets--;
            }else brackets++;
        }
        return (brackets+1)/2;
    }
}